src/legacy/core_plugins/kibana/public/discover/plugin.ts

```diff
@@ -157,13 +159,10 @@ export class DiscoverPlugin implements Plugin<DiscoverSetup, DiscoverStart> {
   private async registerEmbeddable(core: CoreStart, plugins: DiscoverStartPlugins) {
     const { SearchEmbeddableFactory } = await import('./np_ready/embeddable');
     const getInjector = async () => {
-      if (!this.initializeServices) {
-        throw Error('Discover plugin registerEmbeddable:  initializeServices is undefined');
+      if (!this.injectorEmbeddable) {
+        throw Error('Discover plugin registerEmbeddable:  injectorEmbeddable is undefined');
       }
-      await this.initializeServices();
```

Comment:
Since this code is launched only when [createFromSavedObject](https://github.com/elastic/kibana/blob/1ebe7e0039539c72ff7bf0d16e5914ab98af3667/src/legacy/core_plugins/kibana/public/discover/np_ready/embeddable/search_embeddable_factory.ts#L74) function is invoked (for example when Discover saved object is added to the Canvas workpad), I moved it to a separate function `getInjectorEmbeddable` (I guess the name should be changed). And in `start` method before `DocViewsRegistry` creation, an inner module is bootstrapped and an injector is saved in private `injectorEmbeddable` property, that is just used here for embaddables.

As an option we could create one more inner module just for `docViewsRegistry` and leave this removed code.